packer/core.go

```diff
@@ -346,13 +347,24 @@ func (c *Core) validate() error {
 	return err
 }
 
-func (c *Core) init() error {
-	if c.variables == nil {
-		c.variables = make(map[string]string)
+func isDoneInterpolating(v string) (bool, error) {
+	// Check for whether the var contains any more references to `user`, wrapped
+	// in interpolation syntax.
+	filter := "{{\\s*|user\\s*|\\x60?.*\\x60?}}"
```

Comment:
\\x60 represents backticks in the regex. I'm basically making the regex forgive whitespace but otherwise want to see the entire {{ user ``}} format.

Reply:
I think if you used backquotes this could be simplified to :
```suggestion
	filter := `{{\s*|user\s*|\x60?.*\x60?}}`
```

Reply:
This regex will match formats such as  
```
{{ user `bla }}
{{ user bla }}
```
is this expected? 
A regex to avoid that would be:  `{{\s*|user\s*|\x60.*\x60\s*|}}`

Reply:
If this ☝️  is not expected I would add it as a test case to the test you wrote.

Reply:
ah, good catch that's not correct. I need a different test case.